text/0000-mixed-shadow-mode.md

```diff
@@ -0,0 +1,199 @@
+    </x-child>
+</template>
+
+<!-- x-child -->
+<template>
+    <p>child</p>
+</template>
+```
+
+This observable difference will not be remediated as such a change would be non-trivial and
+backwards-incompatible with the current synthetic shadow implementation.
+
+### Lifecycle timing
+
+Due to the way that slotting is implemented in `@lwc/synthetic-shadow`, there is an observable
+difference in the timing of lifecycle hooks for slotted elements.
+
+As mentioned previously, in synthetic mode, slotted elements that are never assigned to a slot are
+not rendered. This means that their lifecycle hooks are never invoked.
+
+Another difference is that for synthetic mode, lifecycle hooks are invoked in the order of
+appearance after they are assigned, whereas in native mode, they are invoked in the order of
+appearance in the template.
+
+### this.shadowRoot instanceof ShadowRoot
+
+LWC components that currently use native shadow roots observe that `this.shadowRoot instanceof
+ShadowRoot` evaluates to `false`. This is due to the fact that the synthetic shadow polyfill patches
+and replaces `ShadowRoot`. This will be remediated by customizing the `instanceof` behavior using
+`Symbol.hasInstance`.
+
+### Listening for non-composed events above the root LWC node
+
+There currently exists logic that allows listeners to handle non-composed events outside of the root
+LWC node if the event originates from a non-LWC component in the subtree. This behavior exists for
+legacy reasons, is only possible to allow in a synthetic Shadow DOM environment, and cannot be
+preserved in a native Shadow DOM context.
+
+### Accessibility
+
+The most common accessibility issue in LWC is related to id-referencing across shadow boundaries
+(e.g., an aria-describedby attribute that references an element outside of its shadow tree). The
+current workaround for this is to bypass the LWC engine's id-mangling by setting attributes
+dynamically, but such a workaround would not work in native Shadow DOM.
+
+### Instrumentation
+
+Applications that rely on instrumentation libraries that don't yet support Shadow DOM are currently
+able to obtain references to descriptors like `addEventListener()` before they are patched and use
+them to override `@lwc/synthetic-shadow` polyfills where needed. Such workarounds would not work for
+events originating from components that prefer native Shadow DOM.
+
+### Styling
+
+Applications currently rely on global stylesheets to apply deeply throughout the DOM. With native
+Shadow DOM, this would not be possible. All of a component's styles would have to be generated and
+added to the component bundle.
+
+## Drawbacks
+
+A potential (non-verified) drawback is that existing components may experience a slight performance
+degradation due to the fact that mixed mode requires logic-forking whenever a Shadow DOM API is
+invoked.
+
+## Alternatives
+
+Other alternatives have not been considered but contributions to this proposal are welcome.

```

Comment:
I think it would be worth spelling out the (somewhat naïve) alternative: just switching from 100% synthetic to 100% native. Here's why I don't think this is an obviously "bad" idea:

- Many folks will wait until the last second to upgrade their components anyway (given human nature), so for them, their experience will basically be the same as switching from 100% synthetic to 100% native.
- We will have to add code to handle mixed mode, and this code will need to be thrown away when we go 100% native.
- There are complexities with supporting mixed mode (native within synthetic, event bubbling and retargeting, styling, etc.), and this could potentially add its own maintenance burden, e.g. if customers start depending on some quirk of native-within-synthetic.

I'm not saying this is what we _should_ be doing, but I think it's a reasonable alternative, so we should have an explanation of why we didn't go this route.

Reply:
Good point, this alternative is worth adding to the RFC. 

I would like to nuance the alternative with the following cons:
- Because Salesforce experiences are composed of components authored by multiple vendors (internal Salesforce devs, customers, ISV, etc), it is nearly impossible to coordinate such kind of switch. 
- Giving a smooth transition path would prevent custom disruption by letting them prepare the native shadow DOM rollout.